app/build.gradle

```diff
@@ -12,8 +12,11 @@ plugins {
 }
 
 unMock {
-    keep "android.net.Uri"
     keep "android.util.Base64"
+    keepStartingWith "libcore."
+    keep "android.net.Uri"
+
+    keepAndRename "java.nio.charset.Charsets" to "xjava.nio.charset.Charsets"
```

Comment:
I'm happy to document this in the build script, but the ChannelSet tests were throwing NoClassDefFoundError for `java.nio.charset.Charsets` and `libcore.net.UriCodec`.  Documentation in [unmock-plugin](https://github.com/bjoernQ/unmock-plugin/blob/master/README.md) indicated that these might be necessary, which resolved the test failures.